src/solve/solver.rs

```diff
@@ -183,12 +183,14 @@ impl Solver {
             match self.cycle_strategy {
                 CycleStrategy::Tabling if slot.cycle => {
                     let actual_answer = result.as_ref().ok().map(|s| s.clone());
-                    if actual_answer == answer {
-                        // Fixed point: break
-                        return result;
-                    } else {
-                        answer = actual_answer;
-                    }
+                    let fixed_point = actual_answer == answer;
+                    match (fixed_point, &actual_answer) {
+                        (_, &Some(Solution::Ambig(Guidance::Unknown))) | (true, _) =>
+                            return result,
+                        _ => ()
+                    };
+
+                    answer = actual_answer;
```

Comment:
Well that's funny because in fact if I provide a test for that and deactivate the fix, the test program will not even load because of the overlap check which will loop forever. So I was indeed relying on this behaviour to test the fix in my PR for #12 (the `trait_wf` test would not load without the fix). Still I added a general test in this PR, which is better for documentation.